plugins/net_plugin/net_plugin.cpp

```diff
@@ -1168,7 +1182,10 @@ namespace eosio {
          connection_ptr c = weak.lock();
          if( !c ) return;
          controller& cc = my_impl->chain_plug->chain();
-         signed_block_ptr sb = cc.fetch_block_by_number( num );
+         signed_block_ptr sb;
+         try {
+            sb = cc.fetch_block_by_number( num );
+         } catch(...) {}
```

Comment:
Would be good to capture reason this failed (at least other than if it is because it is not available in block log) for the ilog below.

Reply:
Yue Han had the same comment I think which I "ignored", so two against one, I'll add something.